src/Microsoft.Health.Dicom.Functions.UnitTests/DurableFunctionsTimedCleanupTests.cs

```diff
@@ -0,0 +1,101 @@
+//using System.Text;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Functions.UnitTests
+{
+    public class DurableFunctionsTimedCleanupTests
+    {
+        private readonly ReindexErrorConfiguration _cleanupConfig;
+        private readonly TimerInfo _timer;
+        private readonly ILogger _logger;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly DateTime _definedNow;
+        private readonly DurableFunctionsTimedCleanup _durableFunctionsCleanup;
+        private readonly IDurableOrchestrationClient _durableOrchestrationClientMock;
+
+        public DurableFunctionsTimedCleanupTests()
+        {
+            _cleanupConfig = new ReindexErrorConfiguration();
+            _cleanupConfig.RuntimeStatuses = new OrchestrationRuntimeStatus[] { OrchestrationRuntimeStatus.Completed };
+            _timer = Substitute.For<TimerInfo>(default, default, default);
+            _logger = Substitute.For<ILogger>();
+            _cancellationTokenSource = new CancellationTokenSource();
+            _definedNow = DateTime.UtcNow;
+            _durableFunctionsCleanup = new DurableFunctionsTimedCleanup(
+                Options.Create(_cleanupConfig),
+                () => _definedNow);
+            _durableOrchestrationClientMock = Substitute.For<IDurableOrchestrationClient>();
+        }
+
+        [Fact]
+        public async Task GivenValidInput_CleanupCompletedDurableFunctions__ShouldPurgeListOfOrchestrationsAsync()
+        {
+            _durableOrchestrationClientMock
+                .ListInstancesAsync(
+                    Arg.Is<OrchestrationStatusQueryCondition>(givenCondition =>
+                        givenCondition.RuntimeStatus.SequenceEqual(_cleanupConfig.RuntimeStatuses)
+                        && givenCondition.CreatedTimeFrom == DateTime.MinValue
+                        && givenCondition.CreatedTimeTo == _definedNow.AddDays(-_cleanupConfig.MinimumAgeDays)
+                    ),
+                    Arg.Any<CancellationToken>())
+                .Returns(Task.FromResult(new OrchestrationStatusQueryResult { DurableOrchestrationState = Array.Empty<DurableOrchestrationStatus>() }));
+
+            await _durableFunctionsCleanup.Run(_timer, _durableOrchestrationClientMock, _logger, _cancellationTokenSource.Token);
+        }
+
+        [Fact]
+        public async Task GivenValidInput_CleanupCompletedDurableFunctions__ShouldPurgeMultiPageListOfOrchestrationsAsync()
+        {
+            IEnumerable<DurableOrchestrationStatus> durableOrchestrationStatuses = new List<DurableOrchestrationStatus>() {
```

Comment:
nit: keep this as `List` on the left-hand side. There's no need to eagerly treat it as an `IEnumerable<T>`